src/imars3d/backend/io/data.py

```diff
@@ -0,0 +1,431 @@
+    ct_fnmatch = param.String(default="*", doc="fnmatch for selecting ct files from ct_dir")
+    ob_fnmatch = param.String(default="*", doc="fnmatch for selecting ob files from ob_dir")
+    dc_fnmatch = param.String(default="*", doc="fnmatch for selecting dc files from dc_dir")
+    # NOTE: 0 means use as many as possible
+    max_workers = param.Integer(default=0, bounds=(0, None), doc="Maximum number of processes allowed during loading")
+
+    def __call__(self, **params):
+        """
+        This makes the class behaves like a function.
+        """
+        # type*bounds check via Parameter
+        _ = self.instance(**params)
+        # sanitize arguments
+        params = param.ParamOverrides(self, params)
+        # type validation is done, now replacing max_worker with an actual integer
+        self.max_workers = multiprocessing.cpu_count() - 2 if params.max_workers == 0 else params.max_workers
+        logger.debug(f"max_worker={self.max_workers}")
+
+        # multiple dispatch
+        # NOTE:
+        #    use set to simplify call signature checking
+        sigs = set([k.split("_")[-1] for k in params.keys() if "fnmatch" not in k])
+        if sigs == {"files", "dir"}:
+            logger.error("Files and dir cannot be used at the same time")
+            raise ValueError("Mix usage of allowed signature.")
+        elif sigs == {"files"}:
+            logger.debug("Load by file list")
+            ct, ob, dc = _load_by_file_list(
+                ct_files=params.get("ct_files"),
+                ob_files=params.get("ob_files"),
+                dc_files=params.get("dc_files", []),  # it is okay to skip dc
+                ct_fnmatch=params.get("ct_fnmatch", "*"),  # incase None got leaked here
+                ob_fnmatch=params.get("ob_fnmatch", "*"),
+                dc_fnmatch=params.get("dc_fnmatch", "*"),
+            )
+            ct_files = params.get("ct_files")
+        elif sigs == {"dir"}:
+            logger.debug("Load by directory")
+            ct_files, ob_files, dc_files = _get_filelist_by_dir(
+                ct_dir=params.get("ct_dir"),
+                ob_dir=params.get("ob_dir"),
+                dc_dir=params.get("dc_dir", []),  # it is okay to skip dc
+                ct_fnmatch=params.get("ct_fnmatch", "*"),  # incase None got leaked here
+                ob_fnmatch=params.get("ob_fnmatch", "*"),
+                dc_fnmatch=params.get("dc_fnmatch", "*"),
+            )
+            ct, ob, dc = _load_by_file_list(
+                ct_files=ct_files,
+                ob_files=ob_files,
+                dc_files=dc_files,
+                ct_fnmatch=params.get("ct_fnmatch", "*"),  # incase None got leaked here
+                ob_fnmatch=params.get("ob_fnmatch", "*"),
+                dc_fnmatch=params.get("dc_fnmatch", "*"),
+            )
+        else:
+            logger.warning("Found unknown input arguments, ignoring.")
```

Comment:
If you end up here, then `ct`, `ob`, and `dc` are undefined and the `return` will raise an exception. Better to catch the problem here, no?